Tweeter/Scripts/app.js

```diff
@@ -20,6 +20,50 @@
     });
 });
 
+var box_of_tweets = {};
+
+var PutTweetsOnPage = function (array_of_tweets) {
+    //clear out old tweets
+    //$("#tweetTarget").html() = "";
+
+    //populate with current tweets
+    for (tweet in array_of_tweets) {
```

Comment:
`for...in...` loops are meant to be on objects - using this syntax on arrays is risky because it can lead to subtle bugs. Use `for...of...` for arrays